frontend/dispatcher.lua

```diff
@@ -617,36 +617,38 @@ arguments are:
 function Dispatcher:execute(ui, settings, gesture)
     for k, v in pairs(settings) do
         if settingsList[k] ~= nil and (settingsList[k].conditions == nil or settingsList[k].conditions == true) then
-            if settingsList[k].category == "none" then
-                if settingsList[k].arg ~= nil then
-                    ui:handleEvent(Event:new(settingsList[k].event, settingsList[k].arg))
-                else
-                    ui:handleEvent(Event:new(settingsList[k].event))
+            if (not settingsList[k].paging and not settingsList[k].rolling) or self.document then
```

Comment:
May be add the comment I suggested in the other PR, so we know what's that about :)
`-- Be sure we don't send a document setting event if there's not yet or not longer a document
`

Reply:
Oups, my bad english, should be "no longer", not "not longer" as @NiLuJe corrected in the other PR :)

Reply:
That's a comment to a previous commit? It reads a bit confusing.

Reply:
I would probably flip the condition for clarity, or is that just me?

Reply:
> That's a comment to a previous commit? It reads a bit confusing.

Initially at https://github.com/koreader/koreader/pull/7691#discussion_r635419091